File: codegen/codegen-erlang/src/test/resources/ir/rest_xml_decode_get_name_request.expected.erl
%% @doc Decode REST-XML request for smithy.beam.demo.http#GetName.
-spec decode_get_name_request(map(), #http_request{}) ->
    {'ok', get_name_input()} | {'error', term()}.
decode_get_name_request(Labels, #http_request{query = Query, headers = Headers, body = Body}) ->
    Name = maps:get(<<"name">>, Labels, undefined),
    {ok, #get_name_input{
        name = Name
    }}.

File: codegen/codegen-elixir/src/test/resources/ir/rest_xml_decode_get_name_request.expected.ex
@spec decode_get_name_request(map(), map()) ::
        {:ok, HttpServiceTypes.GetNameInput.t()} | {:error, term()}
def decode_get_name_request(labels, %{:body => body}) do
  {:ok, %Types.GetNameInput{name: Map.get(labels, "name")}}
end
